InventoryManager/InventoryManager/InventoryManager.cs

```diff
@@ -0,0 +1,409 @@
+﻿namespace InventoryManager {
+    internal class InventoryManager {
+        public static List<ProductDetails> Products = new List<ProductDetails>();
+        public bool IsOperationSuccessful = false;
+
+        readonly string _consoleId = StringFormatter.AddSpaces(nameof(ProductDetails.Id));
+        readonly string _consoleName = StringFormatter.AddSpaces(nameof(ProductDetails.Name));
+        readonly string _consolePrice = StringFormatter.AddSpaces(nameof(ProductDetails.Price));
+        readonly string _consoleQuantity = StringFormatter.AddSpaces(nameof(ProductDetails.Quantity));
+
+        /// <summary>
+        /// Handles display and user selection for a menu, invoking the selected action.
+        /// </summary>
+        /// <param name="menuActions">A list of menu actions to present to the user.</param>
+        public void HandleMenuActions(List<Menu> menuActions, bool isDisplayProduct = false) {
+            do {
+                Console.Clear();
+                if (isDisplayProduct)
+                    DisplayProducts();
+                DisplayMenuOptions(menuActions);
+                Console.Write("\n[Menu] Enter your choice: ");
+
+                string input = Console.ReadLine();
+                bool isNumber = int.TryParse(input, out int choice);
+
+                if (isNumber && choice > 0 && choice <= menuActions.Count) {
+                    if (choice == menuActions.Count)
+                        return;
+                    menuActions[choice - 1].Handler.Invoke();
+                } else {
+                    Console.WriteLine("[Error] Invalid choice!");
+                }
+
+                PromptForContinuation();
+            } while (true);
+        }
+
+        /// <summary>
+        /// Displays the given menu options with an optional title.
```

Comment:
Can we have a default argument alone? 
It should be fine if we have a local variable inside the method.

```suggestion
        /// Displays the given menu options with a title.
```